R/validation.R

```diff
@@ -203,6 +203,63 @@ validate_surv_truth_list_estimate <- function(truth,
       call = call
     )
   }
+
+  all_eval_times_list <- lapply(estimate, function(x) x$.eval_time)
+  all_eval_times <- unlist(all_eval_times_list)
+
+  if (any(is.na(all_eval_times))) {
+    cli::cli_abort(
+      c(
+        x = "Missing values in {.field .eval_time} are not allowed."
+      ),
+      call = call
+    )
+  }
+
+  if (any(all_eval_times < 0)) {
+    offenders <- unique(all_eval_times[all_eval_times < 0])
+
+    cli::cli_abort(
+      c(
+        x = "Negative values of {.field .eval_time} are not allowed.",
+        i = "The following negative values were found: {.val {offenders}}."
+      ),
+      call = call
+    )
+  }
+
+  if (any(is.infinite(all_eval_times))) {
+    cli::cli_abort(
+      c(
+        x = "Infinite values of {.field .eval_time} are not allowed."
+      ),
+      call = call
+    )
+  }
+
+  any_duplicates <- any(
+    vapply(all_eval_times_list, function(x) any(table(x) > 1), logical(1))
+  )
+  if (any_duplicates) {
+    cli::cli_abort(
+      c(
+        x = "Duplicate values of {.field .eval_time} are not allowed."
+      ),
+      call = call
+    )
+  }
+
+  any_not_in_order <- any(
+    vapply(all_eval_times_list, function(x) is.unsorted(x), logical(1))
+  )
+  if (any_not_in_order) {
+    cli::cli_abort(
+      c(
+        x = "Values of {.field .eval_time} must be in increasing order."
+      ),
+      call = call
+    )
+  }
```

Comment:
In tune, we explicitly make use of the order of time points in `eval_time`: for Bayesian optimization and the tuning functions in finetune (racing etc), we can only optimize for one time point so the functions use the first one. If you want to optimize for `10` but also want the metrics to be calculated for `5`, you would set `eval_time = c(10, 5)`. Is the order a requirement from yardstick? If so, we need to decide where the re-ordering should happen. If not, we can just remove this check.

Reply:
lets remove the check + add tests to make sure that out of order `eval_time` works in {yardstick}